python/requirements.txt

```diff
@@ -24,6 +24,7 @@ py-spy >= 0.2.0
 pyyaml
 redis >= 3.5.0
 requests
+pathspec
```

Comment:
We should be really careful about adding new dependencies. Is this strictly necessary? If it is, please tag this PR with api-approval-required and we should also discuss offline

Reply:
> Hmm, it does look like the logic to implement this is quite complex:
> https://github.com/cpburnz/python-path-specification/blob/master/pathspec/patterns/gitwildmatch.py
> 
> Given that the library is very lightweight I think it's probably ok to add it as a dependency
> 
> ```
> (base) ray/serve % pip install -U pathspec
> Requirement already satisfied: pathspec in /Users/eoakes/anaconda3/lib/python3.8/site-packages (0.7.0)
> Collecting pathspec
>   Using cached pathspec-0.8.1-py2.py3-none-any.whl (28 kB)
> Installing collected packages: pathspec
>   Attempting uninstall: pathspec
>     Found existing installation: pathspec 0.7.0
>     Uninstalling pathspec-0.7.0:
>       Successfully uninstalled pathspec-0.7.0
> Successfully installed pathspec-0.8.1
> ```

I think I can make the version of the lib fixed to reduce potential risk.

Reply:
@richardliaw would a fixed version pathspec be acceptable to add to core deps?